README.md
# Flutter Login Template

A Flutter package to create set of pages are used for sign in/up workflow. It will be useful for your awesome app.

![Demo](./demo.gif)

## Example

The most design of Mobile Apps are same same now. Based on them, I created this package to help you to cut off many of developing time. Save them to enjoy a coffee or beer :D.

```dart
import 'package:flutter_login_template/flutter_login_template.dart';

enum _State {
  signIn,
  signUp,
  forgot,
  confirm,
  create,
}

class _MyAppState extends State<MyApp> {
  late LoginTemplateStyle style;
  _State state = _State.signIn;

  @override
  void initState() {
    style = LoginTemplateStyle.defaultTemplate;
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    var logo = Icon(
      Icons.android_rounded,
      size: 80,
    );

    var signInPage = LoginTemplateSignInPage(
      logo: logo,
      style: style,
      onPressedSignIn: () {},
      onPressedSignUp: () {
        setState(() {
          state = _State.signUp;
        });
      },
      onPressedForgot: () {
        setState(() {
          state = _State.forgot;
        });
      },
      socialButtons: [
        LoginTemplateSocialButton(
          text: 'Apple',
          onPressed: () {},
          icon: Icon(
            Icons.account_circle_sharp,
            size: 16,
            color: style.socialButtonTextStyle.color,
          ),
          style: style,
        ),
        LoginTemplateSocialButton(
          text: 'Google',
          onPressed: () {},
          icon: Icon(
            Icons.android,
            size: 16,
            color: style.socialButtonTextStyle.color,
          ),
          style: style,
        )
      ],
      term: LoginTemplateTerm(
        style: style,
        onPressedTermOfService: () {},
        onPressedPrivacyPolicy: () {},
      ),
    );

    var signUpPage = LoginTemplateSignUpPage(
      logo: logo,
      style: style,
      onPressedSignIn: () {
        setState(() {
          state = _State.signIn;
        });
      },
      onPressedSignUp: () {
        setState(() {
          state = _State.confirm;
        });
      },
      term: LoginTemplateTerm(
        style: style,
        onPressedTermOfService: () {},
        onPressedPrivacyPolicy: () {},
      ),
    );

    var forgotPasswordPage = LoginTemplateForgotPasswordPage(
        logo: logo,
        style: style,
        onPressedNext: () {
          setState(() {
            state = _State.confirm;
          });
        });

    var confirmCodePage = LoginTemplateConfirmCodePage(
      logo: logo,
      style: style,
      onPressedNext: () {
        setState(() {
          state = _State.create;
        });
      },
      onPressedResend: () {},
    );

    var createPassword = LoginTemplateCreatePasswordPage(
      logo: logo,
      style: style,
      errorTextPassword:
          'Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.',
      onPressedNext: () {
        setState(() {
          state = _State.signIn;
        });
      },
    );

    Widget body;
    switch (state) {
      case _State.signUp:
        body = signUpPage;
        break;
      case _State.forgot:
        body = forgotPasswordPage;
        break;
      case _State.confirm:
        body = confirmCodePage;
        break;
      case _State.create:
        body = createPassword;
        break;
      case _State.signIn:
      default:
        body = signInPage;
        break;
    }

    return MaterialApp(
      title: 'Example',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        accentColor: Colors.orangeAccent,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text('Flutter Login Template'),
        ),
        body: SingleChildScrollView(
          child: body,
        ),
      ),
    );
  }
}
```

## Development environment

```
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.3.1 21E258 darwin-x64, locale en-VN)
    • Flutter version 3.0.1 at ~/fvm/versions/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (5 days ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at ~/Library/Android/sdk
    • Platform android-31, build-tools 30.0.3
    • ANDROID_HOME = ~/Library/Android/sdk
    • ANDROID_SDK_ROOT = ~/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0
```

## TODO
- [ ] Animations.
- [ ] Flat design.
- [ ] Neumorphism design.
- [ ] Pin code page.
- [ ] Biometric sign in option.
- [ ] Biometric configuration page.
- [ ] User profile page.

CHANGELOG.md
## [1.1.0] - 25/05/2022

* Migrate to flutter 3.0.1.

## [1.0.1+3] - 24/04/2022

* Migrate to flutter stable.
* add option passwordVisible.
* Minor bug fixed: only show errorText if not empty. 
* add option TextField enabled.
* add option maxLines.
* add option signInSectionVisible.

## [1.0.0+2] - 05/09/2021

* Fix method LoginTemplateStyle.only.

## [1.0.0+1] - 25/06/2021

* Fix demo gif.

## [1.0.0] - 03/06/2021

* First release version.

lib/pages/sign_in_page.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_login_template/commons.dart';

/// LoginTemplateSignInPage is used for sign in page.
class LoginTemplateSignInPage extends StatelessWidget {
  /// See [LoginTemplateStyle]
  final LoginTemplateStyle style;

  /// The action of SignIn button.
  final Function() onPressedSignIn;

  /// The action of ForgotPassword button to navigate to [LoginTemplateForgotPasswordPage]
  final Function() onPressedForgot;

  /// The action of SignUp button to navigate to [LoginTemplateSignUpPage]
  final Function() onPressedSignUp;

  /// Your product logo
  final Widget logo;

  /// The text of SignIn button.
  final String buttonTextSignIn;

  /// The text of ForgotPassword button.
  final String buttonTextForgotPassword;

  /// The text of SignUp button.
  final String buttonTextSignUp;

  /// List of Social button lets the users could sign in/up by their social account (Ex: Facebook, AppleId,...).
  final List<LoginTemplateSocialButton>? socialButtons;

  /// See [LoginTemplateTerm]
  final LoginTemplateTerm? term;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final TextInputAction textInputActionUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final TextInputType keyboardTypeUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final String hintTextUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final String? errorTextUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final List<TextInputFormatter>? inputFormattersUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final Function()? onTapUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final TextEditingController? controllerUser;

  /// Use for the TextField lets the users input their password. See also [LoginTemplateTextField]
  final TextInputAction textInputActionPassword;

  /// Use for the TextField lets the users input their password. See also [LoginTemplateTextField]
  final TextInputType keyboardTypePassword;

  /// Use for the TextField lets the users input their password. See also [LoginTemplateTextField]
  final String hintTextPassword;

  /// Use for the TextField lets the users input their password. See also [LoginTemplateTextField]
  final String? errorTextPassword;

  /// Use for the TextField lets the users input their password. See also [LoginTemplateTextField]
  final List<TextInputFormatter>? inputFormattersPassword;

  /// Use for the TextField lets the users input their password. See also [LoginTemplateTextField]
  final Function()? onTapPassword;

  /// Use for the TextField lets the users input their password. See also [LoginTemplateTextField]
  final TextEditingController? controllerPassword;

  final bool passwordVisible;

  const LoginTemplateSignInPage({
    Key? key,
    required this.logo,
    required this.onPressedSignIn,
    required this.onPressedForgot,
    required this.onPressedSignUp,
    required this.style,
    this.textInputActionUser: TextInputAction.next,
    this.keyboardTypeUser: TextInputType.emailAddress,
    this.hintTextUser: 'User name / Email / Phone number',
    this.errorTextUser,
    this.inputFormattersUser,
    this.onTapUser,
    this.controllerUser,
    this.textInputActionPassword: TextInputAction.done,
    this.keyboardTypePassword: TextInputType.visiblePassword,
    this.hintTextPassword: 'Password',
    this.errorTextPassword,
    this.inputFormattersPassword,
    this.onTapPassword,
    this.controllerPassword,
    this.buttonTextSignIn: 'Sign In',
    this.buttonTextForgotPassword: 'Forgot password?',
    this.buttonTextSignUp: 'Sign Up',
    this.socialButtons,
    this.term,
    this.passwordVisible: true,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: style.screenPadding,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Padding(
            padding: EdgeInsets.only(
              top: style.verticalSpacingBetweenGroup,
            ),
            child: Center(
              child: logo,
            ),
          ),
          Padding(
            padding: EdgeInsets.only(
              top: style.screenPadding.top + style.verticalSpacingBetweenGroup,
            ),
            child: LoginTemplateTextField(
              style: style,
              textInputAction: textInputActionUser,
              keyboardType: keyboardTypeUser,
              hintText: hintTextUser,
              controller: controllerUser,
              onTap: onTapUser,
              errorText: errorTextUser,
              inputFormatters: inputFormattersUser,
            ),
          ),
          if (passwordVisible)
            Padding(
              padding:
                  EdgeInsets.only(top: style.verticalSpacingBetweenComponents),
              child: LoginTemplateTextField(
                style: style,
                textInputAction: textInputActionPassword,
                keyboardType: keyboardTypePassword,
                hintText: hintTextPassword,
                controller: controllerPassword,
                onTap: onTapPassword,
                errorText: errorTextPassword,
                inputFormatters: inputFormattersPassword,
              ),
            ),
          Padding(
            padding:
                EdgeInsets.only(top: style.verticalSpacingBetweenComponents),
            child: LoginTemplateButton(
              text: buttonTextSignIn,
              onPressed: onPressedSignIn,
              style: style,
            ),
          ),
          Padding(
            padding: EdgeInsets.only(
              top: style.verticalSpacingBetweenGroup,
            ),
            child: Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                if (passwordVisible)
                  LoginTemplateInlineButton(
                    text: buttonTextForgotPassword,
                    onPressed: onPressedForgot,
                    style: style,
                  ),
                if (passwordVisible)
                  Text(
                    ' | ',
                    style: style.messageTextStyle,
                  ),
                LoginTemplateInlineButton(
                  text: buttonTextSignUp,
                  onPressed: onPressedSignUp,
                  style: style,
                ),
              ],
            ),
          ),
          if (socialButtons != null)
            Padding(
              padding: EdgeInsets.only(
                top: style.verticalSpacingBetweenGroup,
              ),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.stretch,
                children: socialButtons!
                    .map((e) => Padding(
                          padding: EdgeInsets.only(
                            bottom: style.verticalSpacingBetweenComponents,
                          ),
                          child: e,
                        ))
                    .toList(),
              ),
            ),
          if (term != null) term!,
        ],
      ),
    );
  }
}


lib/pages/sign_up_page.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_login_template/commons.dart';

/// LoginTemplateSignUpPage is used for sign up page.
class LoginTemplateSignUpPage extends StatelessWidget {
  /// See [LoginTemplateStyle]
  final LoginTemplateStyle style;

  /// The action of SignUp button.
  final Function() onPressedSignUp;

  /// The action of SignIn button to navigate to [LoginTemplateSignInPage]
  final Function() onPressedSignIn;

  /// Your product logo
  final Widget logo;

  /// The text of SignIn button.
  final String buttonTextSignIn;

  /// The text of SignUp button.
  final String buttonTextSignUp;

  /// See [LoginTemplateTerm]
  final LoginTemplateTerm? term;

  /// The text of SignIn button.
  final String textMessageAlreadyHaveAccount;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final TextInputAction textInputActionUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final TextInputType keyboardTypeUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final String hintTextUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final String? errorTextUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final List<TextInputFormatter>? inputFormattersUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final Function()? onTapUser;

  /// Use for the TextField lets the users input their email or phone number. See also [LoginTemplateTextField]
  final TextEditingController? controllerUser;

  /// Use for the TextField lets the users input their full name. See also [LoginTemplateTextField]
  final TextInputAction textInputActionFullName;

  /// Use for the TextField lets the users input their full name. See also [LoginTemplateTextField]
  final TextInputType keyboardTypeFullName;

  /// Use for the TextField lets the users input their full name. See also [LoginTemplateTextField]
  final String hintTextFullName;

  /// Use for the TextField lets the users input their full name. See also [LoginTemplateTextField]
  final String? errorTextFullName;

  /// Use for the TextField lets the users input their full name. See also [LoginTemplateTextField]
  final List<TextInputFormatter>? inputFormattersFullName;

  /// Use for the TextField lets the users input their full name. See also [LoginTemplateTextField]
  final Function()? onTapFullName;

  /// Use for the TextField lets the users input their full name. See also [LoginTemplateTextField]
  final TextEditingController? controllerFullName;

  final bool signInSectionVisible;

  const LoginTemplateSignUpPage({
    Key? key,
    required this.logo,
    required this.onPressedSignIn,
    required this.onPressedSignUp,
    required this.style,
    this.textInputActionUser: TextInputAction.next,
    this.keyboardTypeUser: TextInputType.emailAddress,
    this.hintTextUser: 'User name / Email / Phone number',
    this.errorTextUser,
    this.inputFormattersUser,
    this.onTapUser,
    this.controllerUser,
    this.textInputActionFullName: TextInputAction.done,
    this.keyboardTypeFullName: TextInputType.name,
    this.hintTextFullName: 'Full name',
    this.errorTextFullName,
    this.inputFormattersFullName,
    this.onTapFullName,
    this.controllerFullName,
    this.buttonTextSignIn: 'Sign In',
    this.buttonTextSignUp: 'Create Account',
    this.term,
    this.textMessageAlreadyHaveAccount: 'Already have an account? ',
    this.signInSectionVisible: true,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: style.screenPadding,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Padding(
            padding: EdgeInsets.only(
              top: style.verticalSpacingBetweenGroup,
            ),
            child: Center(
              child: logo,
            ),
          ),
          Padding(
            padding: EdgeInsets.only(
              top: style.screenPadding.top + style.verticalSpacingBetweenGroup,
            ),
            child: LoginTemplateTextField(
              textInputAction: textInputActionFullName,
              keyboardType: keyboardTypeFullName,
              hintText: hintTextFullName,
              controller: controllerFullName,
              onTap: onTapFullName,
              errorText: errorTextFullName,
              inputFormatters: inputFormattersFullName,
              style: style,
            ),
          ),
          Padding(
            padding:
                EdgeInsets.only(top: style.verticalSpacingBetweenComponents),
            child: LoginTemplateTextField(
              style: style,
              textInputAction: textInputActionUser,
              keyboardType: keyboardTypeUser,
              hintText: hintTextUser,
              controller: controllerUser,
              onTap: onTapUser,
              errorText: errorTextUser,
              inputFormatters: inputFormattersUser,
            ),
          ),
          Padding(
            padding:
                EdgeInsets.only(top: style.verticalSpacingBetweenComponents),
            child: LoginTemplateButton(
              text: buttonTextSignUp,
              onPressed: onPressedSignUp,
              style: style,
            ),
          ),
          if (signInSectionVisible)
            Padding(
              padding: EdgeInsets.only(
                top: style.verticalSpacingBetweenGroup,
              ),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Text(
                    textMessageAlreadyHaveAccount,
                    style: style.messageTextStyle,
                  ),
                  LoginTemplateInlineButton(
                    text: buttonTextSignIn,
                    onPressed: onPressedSignIn,
                    style: style,
                  ),
                ],
              ),
            ),
          if (term != null) term!,
        ],
      ),
    );
  }
}


lib/pages/confirm_code_page.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_login_template/commons.dart';

/// LoginTemplateConfirmCodePage is used to show a page, lets users can input the OTP code sent to their email/phone number.
class LoginTemplateConfirmCodePage extends StatelessWidget {
  /// See [LoginTemplateStyle]
  final LoginTemplateStyle style;

  /// Your product logo
  final Widget logo;

  /// The message is used to explain to users.
  final String textMessage;

  /// The text of button to go to next step.
  final String buttonTextNext;

  /// The action of button to go to next step.
  final Function() onPressedNext;

  /// The message is used to explain to users about resending the OTP code.
  final String textMessageResend;

  /// The text of button to resend the OTP code.
  final String buttonTextResend;

  /// The action of button to resend the OTP code.
  final Function() onPressedResend;

  /// Use for the TextField lets the users input the OTP code. See also [LoginTemplateTextField]
  final TextInputAction textInputActionCode;

  /// Use for the TextField lets the users input the OTP code. See also [LoginTemplateTextField]
  final TextInputType keyboardTypeCode;

  /// Use for the TextField lets the users input the OTP code. See also [LoginTemplateTextField]
  final String hintTextCode;

  /// Use for the TextField lets the users input the OTP code. See also [LoginTemplateTextField]
  final String? errorTextCode;

  /// Use for the TextField lets the users input the OTP code. See also [LoginTemplateTextField]
  final List<TextInputFormatter>? inputFormattersCode;

  /// Use for the TextField lets the users input the OTP code. See also [LoginTemplateTextField]
  final Function()? onTapCode;

  /// Use for the TextField lets the users input the OTP code. See also [LoginTemplateTextField]
  final TextEditingController? controllerCode;

  /// Constructor
  const LoginTemplateConfirmCodePage({
    Key? key,
    required this.logo,
    required this.style,
    this.textMessage: 'Enter the 6 digit code sent to you at ******.',
    this.buttonTextNext: 'Confirm',
    required this.onPressedNext,
    this.textInputActionCode: TextInputAction.done,
    this.keyboardTypeCode: TextInputType.number,
    this.hintTextCode: 'Code',
    this.errorTextCode,
    this.inputFormattersCode,
    this.onTapCode,
    this.controllerCode,
    this.buttonTextResend: 'Resend',
    required this.onPressedResend,
    this.textMessageResend: 'Not received, even at Spam folder?',
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: style.screenPadding,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Padding(
            padding: EdgeInsets.only(
              top: style.verticalSpacingBetweenGroup,
            ),
            child: Center(
              child: logo,
            ),
          ),
          Padding(
            padding: EdgeInsets.only(
              top: style.screenPadding.top + style.verticalSpacingBetweenGroup,
            ),
            child: Text(
              textMessage,
              style: style.messageTextStyle,
            ),
          ),
          Padding(
            padding: EdgeInsets.only(
              top: style.verticalSpacingBetweenGroup,
            ),
            child: LoginTemplateTextField(
              style: style,
              textInputAction: textInputActionCode,
              keyboardType: keyboardTypeCode,
              hintText: hintTextCode,
              controller: controllerCode,
              onTap: onTapCode,
              errorText: errorTextCode,
              inputFormatters: inputFormattersCode,
            ),
          ),
          Padding(
            padding:
                EdgeInsets.only(top: style.verticalSpacingBetweenComponents),
            child: LoginTemplateButton(
              text: buttonTextNext,
              onPressed: onPressedNext,
              style: style,
            ),
          ),
          Padding(
            padding: EdgeInsets.only(
              top: style.verticalSpacingBetweenGroup,
            ),
            child: Wrap(
              crossAxisAlignment: WrapCrossAlignment.center,
              alignment: WrapAlignment.center,
              children: [
                Text(
                  textMessageResend,
                  style: style.messageTextStyle,
                ),
                LoginTemplateInlineButton(
                  text: buttonTextResend,
                  onPressed: onPressedResend,
                  style: style,
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}


lib/pages/forgot_password_page.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_login_template/commons.dart';

/// LoginTemplateForgotPasswordPage is used to show a page, lets the users could create their password when they forgot it.
class LoginTemplateForgotPasswordPage extends StatelessWidget {
  /// See [LoginTemplateStyle]
  final LoginTemplateStyle style;

  /// Your product logo
  final Widget logo;

  /// The message is used to explain to users.
  final String textMessage;

  /// The text of button to go to next step.
  final String buttonTextNext;

  /// The action of button to go to next step.
  final Function() onPressedNext;

  /// Use for the TextField lets the users input their email or phone number to confirm for creating new password. See also [LoginTemplateTextField]
  final TextInputAction textInputActionUser;

  /// Use for the TextField lets the users input their email or phone number to confirm for creating new password. See also [LoginTemplateTextField]
  final TextInputType keyboardTypeUser;

  /// Use for the TextField lets the users input their email or phone number to confirm for creating new password. See also [LoginTemplateTextField]
  final String hintTextUser;

  /// Use for the TextField lets the users input their email or phone number to confirm for creating new password. See also [LoginTemplateTextField]
  final String? errorTextUser;

  /// Use for the TextField lets the users input their email or phone number to confirm for creating new password. See also [LoginTemplateTextField]
  final List<TextInputFormatter>? inputFormattersUser;

  /// Use for the TextField lets the users input their email or phone number to confirm for creating new password. See also [LoginTemplateTextField]
  final Function()? onTapUser;

  /// Use for the TextField lets the users input their email or phone number to confirm for creating new password. See also [LoginTemplateTextField]
  final TextEditingController? controllerUser;

  /// Constructor
  const LoginTemplateForgotPasswordPage({
    Key? key,
    required this.logo,
    required this.style,
    this.buttonTextNext: 'Send',
    required this.onPressedNext,
    this.textInputActionUser: TextInputAction.next,
    this.keyboardTypeUser: TextInputType.emailAddress,
    this.hintTextUser: 'Email / Phone number',
    this.errorTextUser,
    this.inputFormattersUser,
    this.onTapUser,
    this.controllerUser,
    this.textMessage:
        'Enter the email or phone number associated with your account and we will send an introductions to reset your password.',
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: style.screenPadding,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Padding(
            padding: EdgeInsets.only(
              top: style.verticalSpacingBetweenGroup,
            ),
            child: Center(
              child: logo,
            ),
          ),
          Padding(
            padding: EdgeInsets.only(
              top: style.screenPadding.top + style.verticalSpacingBetweenGroup,
            ),
            child: Text(
              textMessage,
              style: style.messageTextStyle,
            ),
          ),
          Padding(
            padding: EdgeInsets.only(
              top: style.verticalSpacingBetweenGroup,
            ),
            child: LoginTemplateTextField(
              style: style,
              textInputAction: textInputActionUser,
              keyboardType: keyboardTypeUser,
              hintText: hintTextUser,
              controller: controllerUser,
              onTap: onTapUser,
              errorText: errorTextUser,
              inputFormatters: inputFormattersUser,
            ),
          ),
          Padding(
            padding:
                EdgeInsets.only(top: style.verticalSpacingBetweenComponents),
            child: LoginTemplateButton(
              text: buttonTextNext,
              onPressed: onPressedNext,
              style: style,
            ),
          ),
        ],
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_login_template/flutter_login_template.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

enum _State {
  signIn,
  signUp,
  forgot,
  confirm,
  create,
}

class _MyAppState extends State<MyApp> {
  late LoginTemplateStyle style;
  _State state = _State.signIn;

  @override
  void initState() {
    style = LoginTemplateStyle.defaultTemplate;
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    var logo = Icon(
      Icons.android_rounded,
      size: 80,
    );

    var signInPage = LoginTemplateSignInPage(
      logo: logo,
      style: style,
      onPressedSignIn: () {},
      onPressedSignUp: () {
        setState(() {
          state = _State.signUp;
        });
      },
      onPressedForgot: () {
        setState(() {
          state = _State.forgot;
        });
      },
      ///Canceling Logging in by Google/Apple
      // socialButtons: [
      //   LoginTemplateSocialButton(
      //     text: 'Apple',
      //     onPressed: () {},
      //     icon: Icon(
      //       Icons.account_circle_sharp,
      //       size: 16,
      //       color: style.socialButtonTextStyle.color,
      //     ),
      //     style: style,
      //   ),
      //   LoginTemplateSocialButton(
      //     text: 'Google',
      //     onPressed: () {},
      //     icon: Icon(
      //       Icons.android,
      //       size: 16,
      //       color: style.socialButtonTextStyle.color,
      //     ),
      //     style: style,
      //   )
      // ],
      term: LoginTemplateTerm(
        style: style,
        onPressedTermOfService: () {},
        onPressedPrivacyPolicy: () {},
      ),
    );

    var signUpPage = LoginTemplateSignUpPage(
      logo: logo,
      style: style,
      onPressedSignIn: () {
        setState(() {
          state = _State.signIn;
        });
      },
      onPressedSignUp: () {
        setState(() {
          state = _State.confirm;
        });
      },
      term: LoginTemplateTerm(
        style: style,
        onPressedTermOfService: () {},
        onPressedPrivacyPolicy: () {},
      ),
    );

    var forgotPasswordPage = LoginTemplateForgotPasswordPage(
        logo: logo,
        style: style,
        onPressedNext: () {
          setState(() {
            state = _State.confirm;
          });
        });

    var confirmCodePage = LoginTemplateConfirmCodePage(
      logo: logo,
      style: style,
      onPressedNext: () {
        setState(() {
          state = _State.create;
        });
      },
      onPressedResend: () {},
    );

    var createPassword = LoginTemplateCreatePasswordPage(
      logo: logo,
      style: style,
      errorTextPassword:
          'Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.',
      onPressedNext: () {
        setState(() {
          state = _State.signIn;
        });
      },
    );

    Widget body;
    switch (state) {
      case _State.signUp:
        body = signUpPage;
        break;
      case _State.forgot:
        body = forgotPasswordPage;
        break;
      case _State.confirm:
        body = confirmCodePage;
        break;
      case _State.create:
        body = createPassword;
        break;
      case _State.signIn:
      default:
        body = signInPage;
        break;
    }

    return MaterialApp(
      title: 'Example',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSwatch(primarySwatch: Colors.blue)
            .copyWith(secondary: Colors.orangeAccent),
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text('Flutter Login Template'),
        ),
        body: SingleChildScrollView(
          child: body,
        ),
      ),
    );
  }
}
